datacommons_client/endpoints/node.py

```diff
@@ -252,99 +253,233 @@ def fetch_entity_names(
 
     return names
 
+  def _fetch_contained_in_place(
+      self,
+      node_dcids: str | list[str],
+      out: bool = True,
+      contained_type: Optional[str] = None,
+      as_dict: bool = False,
+  ) -> dict[str, list[Node | dict]]:
+    """Fetches places that contain or are contained in the given nodes. Uses the
+          `containedInPlace` property to fetch parent or child place relationships.
+
+        Args:
+            node_dcids (str | list[str]): One or more DCIDs representing geographic places.
+            out (bool, optional): If True, fetch places contained in the given node(s).
+                If False, fetch places that contain the given node(s). Defaults to True.
+            contained_type (str, optional): Optional type constraint (e.g., 'Country',
+                'Country'). If provided, only fetches places of that type.
+            as_dict (bool, optional): If True, returns the result as a dictionary of
+                lists of dictionaries. If False, returns Node objects. Defaults to False.
+
+        Returns:
+            dict[str, list[dict]] | dict[str, list[Any]]: A dictionary where keys are DCIDs
+            and values are lists of place relationships, either as raw objects or
+            dictionaries (if `as_dict` is True).
+        """
+    data = self.fetch_property_values(
+        node_dcids=node_dcids,
+        properties=("containedInPlace+"
+                    if contained_type else "containedInPlace"),
+        out=out,
+        constraints=f"typeOf:{contained_type}" if contained_type else None,
+    ).get_properties()
+
+    if as_dict:
+      for entity, nodes in data.items():
+        if not nodes:
+          continue
+        if isinstance(nodes, Node):
+          return {entity: [nodes.to_dict()]}
+        else:
+          return {entity: [node.to_dict() for node in nodes]}
+    return data
```

Comment:
```suggestion
  if out and contained_type:
    raise ValueError("When 'out' is True, `contained_type' must be None.")
  
  prop = "containedInPlace+" if contained_type else "containedInPlace"
  data = self.fetch_property_values(
        node_dcids=node_dcids,
        properties=prop,
        out=out,
        constraints=f"typeOf:{contained_type}" if contained_type else None,
    ).get_properties()
    
  resp = {}
  for entity, property_nodes in data.items():
    nodes = property_nodes.get(prop, [])
    resp[entity] = [node.to_dict() for node in nodes] if as_dict else nodes
  return resp
```

Reply:
Thank you for fixing the code to work with the latest!